### ISSUE TITLE
Incorrectly reported `TS2322` error on array items after spread

### ISSUE THREAD

**Issue by isti115** [microsoft/TypeScript#55340]
### 🔎 Search Terms

"array spread", "TS2322"

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about 

### ⏯ Playground Link

https://www.typescriptlang.org/play?#code/MYewdgzgLgBAhgLhgQQE6rgTwDxgK4C2ARgKaowA+M0qAlmAOYB8MAvDANoC6A3AFChIsIkjQYc+YmRbsOAOgVwANDACMvIA

### 💻 Code

```ts
const a: Array<number | string> = [];
const b: Array<number> = [...a, 1];
```

### 🙁 Actual behavior

The compiler gives the following two errors, of which only one is appropriate:
```
~/prog/typescript ❯ tsc ./array-spread-error.ts
array-spread-error.ts:2:27 - error TS2322: Type 'string | number' is not assignable to type 'number'.
  Type 'string' is not assignable to type 'number'.

2 const b: Array<number> = [...a, 1];
                            ~~~~

array-spread-error.ts:2:33 - error TS2322: Type 'string | number' is not assignable to type 'number'.
  Type 'string' is not assignable to type 'number'.

2 const b: Array<number> = [...a, 1];
                                  ~
```

### 🙂 Expected behavior

The spread is clearly incorrect, as the original array could potentially contain strings as well, but the number literal should not be marked with the same error.

**Comment by Andarist** [comment]
This one is already being fixed by https://github.com/microsoft/TypeScript/pull/52850 . I pushed out a new test case there that is based on this issue since I originally prepared it with object-oriented cases in mind.
